types/node-vault-client/index.d.ts

```diff
@@ -0,0 +1,377 @@
+     * times with the same name will return the same instance.
+     *
+     * @param {String} name - Vault instance name
+     * @return Vault
+     */
+    static get(name: string): any;
+    /**
+     * Clear named Vault instance
+     *
+     * If no name passed all named instances will be cleared.
+     *
+     * @param {String} [name] - Vault instance name, all instances will be cleared if no name were passed
+     */
+    static clear(name?: string): void;
+    /**
+     * Client constructor function.
+     *
+     * @param {Object} options
+     * @param {Object} options.api
+     * @param {String} options.api.url - the url of the vault server
+     * @param {String} [options.api.apiVersion='v1']
+     * @param {Object} options.auth
+     * @param {String} options.auth.type
+     * @param {Object} options.auth.config - auth configuration variables
+     * @param {Object|false} options.logger - Logger that supports "error", "info", "warn", "trace", "debug" methods. Uses `console` by default. Pass `false` to disable logging.
+     */
+    constructor(options: {
+        api: {
+            url: string;
+            apiVersion?: string;
+        };
+        auth: {
+            type: string;
+            config: any;
+        };
+        logger: any;
+    });
+    __log: any;
+    __api: VaultApiClient;
+    /** */
+    private __auth;
+    __namespace: any;
+    /**
+     * @param {Object} authConfig
+     * @param {string} authConfig.type
+     * @param {string} authConfig.mount
+     * @param {Object} authConfig.config
+     * @param {VaultApiClient} api
+     * @return {VaultBaseAuth}
+     */
+    private __getAuthProvider;
+    /**
+     * Populates Vault's values to NPM "config" module
+     */
+    fillNodeConfig(): any;
+    getHeaders(token: any): {
+        "X-Vault-Token": any;
+        "X-Vault-Namespace": any;
+    } | {
+        "X-Vault-Token": any;
+        "X-Vault-Namespace"?: undefined;
+    };
+    /**
+     * Read secret from Vault
+     * @param {string} path - path to the secret
+     * @returns {Promise<Lease>}
+     */
+    read(path: string): Promise<Lease>;
+    /**
+     * Retrieves secrets list
+     *
+     * @param {string} path - path to the secret
+     * @returns {Promise<Lease>}
+     */
+    list(path: string): Promise<Lease>;
+    /**
+     * Writes data to Vault
+     *
+     * @param path - path used to write data
+     * @param {object} data - data to write
+     * @returns {Promise<T | never>}
+     */
+    write(path: any, data: object): Promise<any>;
+    private __setupLogger;
+}
+
+declare class VaultApiClient {
+    /**
+     * @param {Object} config
+     * @param {String} config.url - the url of the vault server
+     * @param {String} [config.apiVersion='v1']
+     * @param {Object} logger
+     */
+    constructor(config: {
+        url: string;
+        apiVersion?: string;
+    }, logger: any);
+    __config: any;
```

Comment:
I'm not seeing docs for any of these underscore-named fields. Are they intended to be private? We typically don't put these on DT or anything

Reply:
Hi @jakebailey - removed private properties.